### PR TITLE
feat: logn,logout,signup

### PR DESCRIPTION
**Pull request by mimansha-swarup**
- login
- logout
- signup

### CODE REVIEW

File: src/Context/authContext.js [hunk 1]
```diff
@@ -1,10 +1,93 @@
...
 
 export const useAuth = () => useContext(authContext);
 
-export const AuthProvider = ({children}) =>{
-  
-  return( <authContext.Provider>{children}</authContext.Provider>)
-}
+export const AuthProvider = ({ children }) => {
+  const [authState, authDispatch] = useReducer(authReducer, initialState);
+  console.log("authState", authState);
```
Comment (manojasarsa) [inline src/Context/authContext.js]: You can remove these console logs

**Review by manojasarsa — APPROVED** [review]
Good to merge!

File: src/Context/authContext.js [hunk 2]
```diff
@@ -1,10 +1,93 @@
...
+  const navigate = useNavigate();
+
+  const Login = async (email, password) => {
+    try {
+      const response = await axios.post(loginPostApi, { email, password });
```
Comment (BelideNikhil) [inline src/Context/authContext.js:24]: very clever implementation 🔥

File: src/Page/Login/LoginPage.js [hunk 3]
```diff
@@ -2,27 +2,32 @@ import "./LoginPage.css";
...
-    setCredential({email:emailDetails,password:passwordDetails})
-    
-    // TODO: - Authentiaction Part
-  }
+
+    const emailDetails = event.target.email.value;
+    const passwordDetails = event.target.password.value;
+    console.log(emailDetails, passwordDetails);
+    Login(emailDetails, passwordDetails);
+  };
```
Comment (BelideNikhil) [inline src/Page/Login/LoginPage.js:22]: very very clever 🔥🔥

**Review by BelideNikhil — APPROVED** [review]
The UI is really great.
The code is  clean and very cleverly written especially the form handling
You can remove the console logs and you are GTG!

**Pull request merged**